com.unity.multiplayer.mlapi/Runtime/SceneManagement/NetworkSceneManager.cs

```diff
@@ -39,6 +56,22 @@ public class NetworkSceneManager
         /// </summary>
         public event SceneSwitchStartedDelegate OnSceneSwitchStarted;
 
+        /// <summary>
+        /// Event that is invoked on the server when a client completes scene transition
+        /// </summary>
+        public event OnClientLoadedSceneDelegate OnClientLoadedScene;
+
+        /// <summary>
+        /// Event that is invoked on the server when all clients have reported that they have completed scene transition
+        /// </summary>
+        public event OnAllClientsLoadedSceneDelegate OnAllClientsLoadedScene;
+
+        /// <summary>
+        /// Event that is invoked on the clients after all clients have successfully completed scene transition or timed out.
+        /// This event relies on MessageSender, which doesn't send events from the server to itself (which is the case for a Host client).
```

Comment:
I'd put this in a `<remarks>` section.
Also, this limitation is a real shame. Any time someone has to implement a separate path for client-host vs. client because of a framework limitation it feels like we've got a framework problem, not a "it's by design" situation. Maybe talk to the team about getting something on the backlog to fix this?

Reply:
This pattern I was hoping to fix this in the up-coming NetworkSceneManager overhaul.